pinot-core/src/main/java/org/apache/pinot/core/operator/combine/BaseSingleBlockCombineOperator.java

```diff
@@ -71,11 +68,8 @@ protected BaseResultsBlock getNextBlock() {
     try {
       startProcess();
       mergedBlock = mergeResults();
-    } catch (InterruptedException | EarlyTerminationException e) {
-      Exception killedErrorMsg = Tracing.getThreadAccountant().getErrorStatus();
-      throw new QueryCancelledException(
-          "Cancelled while merging results blocks" + (killedErrorMsg == null ? StringUtils.EMPTY
-              : " " + killedErrorMsg), e);
+    } catch (InterruptedException e) {
+      throw new EarlyTerminationException();
```

Comment:
plz include the original exception ?
```suggestion
      throw new EarlyTerminationException(e);
```